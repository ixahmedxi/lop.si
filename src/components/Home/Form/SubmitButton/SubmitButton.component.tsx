/* @jsx jsx */
import { useNeuBoxShadow } from '@hooks/useBoxShadow'
import { FiArrowRight } from 'react-icons/fi'
import { jsx, Spinner, SxStyleProp } from 'theme-ui'

interface SubmitButtonProps {
  isLoading: boolean
}

export const SubmitButton = ({ isLoading }: SubmitButtonProps) => {
  const styles: SxStyleProp = {
    button: {
      height: [5, 5, 6],
      width: [5, 5, 6],
      outline: 'none',
      border: 'none',
      fontSize: [1, 1, '20px'],
      bg: 'primary',
      color: 'background',
      borderRadius: 100,
      p: 0,
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      cursor: 'pointer',
      transition: 'box-shadow 0.2s ease-out, background-color 0.2s ease-out, color 0.2s ease-out',
      ...useNeuBoxShadow(5, 10)
    }
  }

  return (
    <button sx={styles.button} type="submit" aria-label="create a short url">
      {isLoading ? (
        <Spinner variant="styles.spinner" size={24} data-testid="spinner" />
      ) : (
        <FiArrowRight data-testid="arrow-right" />
      )}
    </button>
  )
}
