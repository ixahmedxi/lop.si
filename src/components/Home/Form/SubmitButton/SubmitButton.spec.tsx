import { render } from 'test-utils'
import { SubmitButton } from './SubmitButton.component'

describe('Submit Button', () => {
  it('matches snapshot', () => {
    const { container } = render(<SubmitButton isLoading={false} />)
    expect(container).toMatchSnapshot()
  })

  it('renders spinner when isLoading is true', () => {
    const { queryByTestId, getByTestId } = render(<SubmitButton isLoading={true} />)
    getByTestId('spinner')
    expect(queryByTestId('arrow-right')).toBeNull()
  })

  it('renders arrow when isLoading is false', () => {
    const { queryByTestId, getByTestId } = render(<SubmitButton isLoading={false} />)
    getByTestId('arrow-right')
    expect(queryByTestId('spinner')).toBeNull()
  })
})
