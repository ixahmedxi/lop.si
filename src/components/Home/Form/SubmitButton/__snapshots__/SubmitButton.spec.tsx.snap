// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit Button matches snapshot 1`] = `
.emotion-0 {
  height: 40px;
  width: 40px;
  outline: none;
  border: none;
  font-size: 16px;
  background-color: var(--theme-ui-colors-primary,#8093f2);
  color: var(--theme-ui-colors-background,#2B2D32);
  border-radius: 100px;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-transition: none;
  transition: none;
  box-shadow: none;
}

@media screen and (min-width:576px) {
  .emotion-0 {
    height: 40px;
    width: 40px;
    font-size: 16px;
  }
}

@media screen and (min-width:768px) {
  .emotion-0 {
    height: 48px;
    width: 48px;
    font-size: 20px;
  }
}

<div>
  <button
    aria-label="create a short url"
    class="emotion-0"
    type="submit"
  >
    <svg
      data-testid="arrow-right"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 5 19 12 12 19"
      />
    </svg>
  </button>
</div>
`;
